packages/imperative/__tests__/config/cmd/report-env/EnvQuery.unit.test.ts

```diff
@@ -0,0 +1,397 @@
+                        }
+                    },
+                    mProperties: {
+                        defaults: {
+                            base: "fakeBaseProfNm",
+                            zosmf: "fakeZosmfProfNm",
+                            jclcheck: "fakeJclCheckProfNm",
+                            tso: "fakeTsoProfNm",
+                            cics: "fakeCicsProfNm"
+                        }
+                    },
+                    layers: [{
+                        exists: true,
+                        global: true,
+                        user: false,
+                        path: "fakeDir/zowe.config.json",
+                        properties: {
+                            profiles: {
+                                fakeProf: {
+                                    secure: ["fakeSecureName"],
+                                    properties: {fakePropNm: "fakePropVal"}
+                                }
+                            }
+                        }
+                    }],
+                    api: {
+                        secure: {
+                            secureFields: jest.fn(() => { return []; })
+                        }
+                    }
+                };
+            })
+        });
+        Object.defineProperty(impCfg, "cliHome", {
+            configurable: true,
+            get: jest.fn(() => {
+                return fakeCliHomeDir;
+            })
+        });
+    });
+
+    describe("test getEnvItemVal function", () => {
+        it("should report the zowe version", async () => {
+            const itemObj: IGetItemVal = await EnvQuery.getEnvItemVal(ItemId.ZOWE_VER);
+            expect(itemObj.itemVal).toMatch(/[0-9]+.[0-9]+.[0-9]+/);
+            expect(itemObj.itemValMsg).toContain("Zowe CLI version =");
+            expect(itemObj.itemProbMsg).toBe("");
+        });
+
+        it("should report the NodeJs version", async () => {
+            const itemObj: IGetItemVal = await EnvQuery.getEnvItemVal(ItemId.NODEJS_VER);
+            expect(itemObj.itemVal).toMatch(/[0-9]+.[0-9]+.[0-9]+/);
+            expect(itemObj.itemValMsg).toContain("Node.js version =");
+            expect(itemObj.itemProbMsg).toBe("");
+        });
+
+        it("should report the NVM version", async () => {
+            const itemObj: IGetItemVal = await EnvQuery.getEnvItemVal(ItemId.NVM_VER);
+            if (!itemObj.itemVal.includes("nvm failed to display any output")) {
+                expect(itemObj.itemVal).toMatch(/[0-9]+.[0-9]+.[0-9]+/);
+                expect(itemObj.itemValMsg).toContain("Node Version Manager version =");
+            }
+            expect(itemObj.itemProbMsg).toBe("");
+        });
+
+        it("should report the platform", async () => {
+            const itemObj: IGetItemVal = await EnvQuery.getEnvItemVal(ItemId.PLATFORM);
+            expect(itemObj.itemVal === "win32" || itemObj.itemVal === "linux" || itemObj.itemVal === "darwin").toBeTruthy();
```

Comment:
I agree that it would easy to add tests for extra platforms. My feeling is that we should not test for platforms that we do not officially support. The test could then be a bit misleading. I suggest that we add a test for a platform, once we decide to support a new platform.